engine/src/main/java/org/hibernate/validator/constraints/UniqueItems.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.constraints;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import javax.validation.Constraint;
+import javax.validation.Payload;
+
+import org.hibernate.validator.internal.constraintvalidators.hv.UniqueItemsValidator;
+
+/**
+ * Validates that the provided Collection is a Set of unique items. By default, uniqueness is defined using the object's
```

Comment:
"is a Set of" seems a bit confusing because it's not limited to `Set`, the Java type. How about "Validates that the provided {@link Collection} contains only unique elements."?

Is it limited to `Collection` actually, or could it even be `Iterable`?

Reply:
I tried to use many synonyms to help clarify behavior for different users, and outside of the Java concept of a set, we are essentially attempting to verify a set- or bag-like behavior. But I'm happy to use the terminology you've provided if you think it fits better with the project.

Reply:
It can be iterable, but it's substantially more expensive for large collections, since iterables don't provide a size method to quickly check if the size of the set of items equals the size of the original collection.
Also, since iterables tend to be streams, the annotation would have to read the whole stream to check, which would add complexity and might not be the behavior the user expects.